src/filesystem/implementation.js

```diff
@@ -1870,11 +1870,24 @@ function appendFile(fs, context, path, data, options, callback) {
   if(!flags) {
     return callback(new Errors.EINVAL('flags is not valid', path));
   }
+  
+  if (typeof options === 'object') {
+    if (options.encoding === undefined) {
```

Comment:
Prefer a check for `!options.*` for all of these, so we also deal with things that aren't explicitly `undefined`

```suggestion
    if (!options.encoding) {
```